gcm/response.go

```diff
@@ -0,0 +1,43 @@
+package gcm
+
+import (
+	"fmt"
+)
+
+type ErrorResponse struct {
+	ErrCode    string // todo GCMErrorResponseCode
+	HttpStatus int
+}
+
+func (er ErrorResponse) Error() string {
+	return fmt.Sprintf("%s:[http_status:%d]", er.ErrCode, er.HttpStatus)
+}
+
+// Response is the gcm connection server response
+type Response struct {
+	Header ResponseHeader `json:"header"`
+	Body   ResponseBody   `json:"body"`
+}
+
+// ResponseHeader gcm response header
+type ResponseHeader struct {
+	StatusCode int `json:"status_code"`
+}
+
+// ResponseBody gcm response body
+type ResponseBody struct {
+	MulticastID  int      `json:"multicast_id"`
+	Success      int      `json:"success"`
+	Failure      int      `json:"failure"`
+	CanonicalIDs int      `json:"canonical_ids"`
```

Comment:
multicast_id, success, failure and cannonical_ids is required.
https://developers.google.com/cloud-messaging/http-server-ref#table5